### ISSUE TITLE
`StructuredGrid` not showing using `ipygany` backend

### ISSUE THREAD

**Issue by dstansby** [pyvista/pyvista#1530]
**Describe the bug, what's wrong, and what you expect:**

Trying to plot a `StructuredGrid` mesh using the `ipygany` backend results in an empty plotter with no mesh shown. I'm not sure whether this is because `ipygany` doesn't support plotting `StructuredGrid` meshes, in which case it would be good to throw an error or warning when the user tries to do this. I couldn't find anything in the docs about this though.

-----

**To Reproduce**

```py
.. jupyter-execute::
   import pyvista
   pyvista.set_jupyter_backend('ipygany')


  import numpy as np

  plotter = pyvista.Plotter()
  xrng = np.arange(-10, 10, 2)
  yrng = np.arange(-10, 10, 2)
  zrng = np.arange(-10, 10, 2)
  x, y, z = np.meshgrid(xrng, yrng, zrng)
  grid = pyvista.StructuredGrid(x, y, z)
  plotter.add_mesh(grid)
  plotter.show(jupyter_backend='ipygany')
```

**Screenshots**

-----

**System Information:**

```txt
--------------------------------------------------------------------------------
  Date: Tue Jul 20 11:56:10 2021 BST

                OS : Darwin
            CPU(s) : 4
           Machine : x86_64
      Architecture : 64bit
               RAM : 8.0 GiB
       Environment : Python
        GPU Vendor : Intel Inc.
      GPU Renderer : Intel Iris OpenGL Engine
       GPU Version : 4.1 INTEL-16.1.11

  Python 3.9.5 (default, May 18 2021, 12:31:01)  [Clang 10.0.0 ]

           pyvista : 0.31.3
               vtk : 9.0.3
             numpy : 1.20.2
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.7
            meshio : 4.4.6
        matplotlib : 3.4.2
             PyQt5 : 5.15.2
           IPython : 7.25.0
             scipy : 1.5.4
              tqdm : 4.54.0

  Intel(R) oneAPI Math Kernel Library Version 2021.2-Product Build 20210312
  for Intel(R) 64 architecture applications
--------------------------------------------------------------------------------

```

**Comment by akaszynski** [comment]
Good point!  Here's what I'd do to get around this:

You're quite right, `ipygany` doesn't support structured grids out of the box.  Here's a way around it at the moment until we get this patched:

```py
import pyvista
import numpy as np

plotter = pyvista.Plotter()
xrng = np.arange(-10, 10, 2)
yrng = np.arange(-10, 10, 2)
zrng = np.arange(-10, 10, 2)
x, y, z = np.meshgrid(xrng, yrng, zrng)
grid = pyvista.StructuredGrid(x, y, z)
plotter.add_mesh(grid.extract_surface())
plotter.show()
```

**Comment by banesullivan** [comment]
I think this was fixed by: #1557 and is handled here:

https://github.com/pyvista/pyvista/blob/fd14e16f22db6d485a8bfe4101b39bbcb83a88c6/pyvista/jupyter/pv_ipygany.py#L85-L86
